app/javascript/components/charts/ChartDoughnut.vue

```diff
@@ -0,0 +1,93 @@
+<template>
+  <div class="chart--doughnut">
+    <h5 class="chart--doughnut__title">
+      {{ title }}
+    </h5>
+    <template v-if="Object.entries(this.statistics).length > 0">
+      <canvas
+        :id="chartId"
+        class="chart--doughnut__chartarea"
+      />
+    </template>
+    <template v-else>
+      <p>No data currently available for this property</p>
+    </template>
+  </div>
+</template>
+
+<script>
+import Chart from 'chart.js'
+
+export default {
+  name: 'ChartDoughnut',
+  props: {
+    chartId: {
+      type: String,
+      required: true
+    },
+    statistics: {
+      type: Object,
+      default: () => ({})
+    },
+    title: {
+      type: String,
+      default: ''
+    }
+  },
+  data() {
+    return {
+      formattedStats: undefined,
+      datasetOptions: {
+        backgroundColors: [
+          'rgba(255, 0, 0, 0.5)',
+          'rgba(0, 255, 0, 0.5)',
+          'rgba(0, 0, 255, 0.5)',
+          'rgba(255, 255, 0, 0.5)'
+        ]
+      },
+      generalOptions: {
+        responsive: false
+      }
+    }
+  },
+  mounted() {
+    this.customiseDataset()
+    this.createChart(this.chartId)
+  },
+  methods: {
+    createChart(chartId) {
+      const chartElem = document.getElementById(chartId)
+      
+      return new Chart(chartElem, {
+        type: 'doughnut',
+        data: this.formattedStats,
+        options: this.generalOptions
+      })
+    },
+    setStyleOptions() {
```

Comment:
It is setting style options though because it's setting the background colour and border rather than getting them from somewhere else

Reply:
It returns an object with the styling options, but these options are not actually 'set' until line 81. Or at least that's how I would look at this. 

If you were to call this method in customiseDataset and do nothing else, then the styling options wouldn't have been set. They're only set because of the logic where you spread them into to formattedStats. So I see it as customiseDataset gets the styleOptions and then sets them in formattedStats.